flow/connectors/snowflake/qrep_avro_sync.go

```diff
@@ -138,28 +139,15 @@ func (s *SnowflakeAvroSyncMethod) SyncQRepRecords(
 		return 0, err
 	}
 
-	numRecords, localFilePath, err := s.writeToAvroFile(stream, avroSchema, partition.PartitionId, config.FlowJobName)
+	avroFile, err := s.writeToAvroFile(stream, avroSchema, partition.PartitionId, config.FlowJobName)
 	if err != nil {
 		return 0, err
 	}
-
-	if localFilePath != "" {
-		defer func() {
-			log.Infof("removing temp file %s", localFilePath)
-			err := os.Remove(localFilePath)
-			if err != nil {
-				log.WithFields(log.Fields{
-					"flowName":         config.FlowJobName,
-					"partitionID":      partition.PartitionId,
-					"destinationTable": dstTableName,
-				}).Errorf("failed to remove temp file %s: %v", localFilePath, err)
-			}
-		}()
-	}
+	defer avroFile.Cleanup()
```

Comment:
can there be an error in the middle of writing the file and lead to a file still being left around?
in that case we'll return the error, but not cleanup.

Similar cases below